src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/EngineIsolationTest.java

```diff
@@ -562,6 +562,11 @@ protected DatabaseEngine[] prepareEngineForDeadlockRecoveryTest() throws Excepti
         engines[0] = DatabaseFactory.getConnection(properties);
         final EntityEntry entry = entry().set("COL1", 1).set("COL2", 0).build();
 
+        if (this.engineDriver.equals(DatabaseEngineDriver.ORACLE)) {
+            // Fix ORA-00604: error occurred at recursive SQL level 1ORA-08177: can't serialize access for this transaction
+            engines[0].getConnection().setTransactionIsolation(Connection.TRANSACTION_READ_COMMITTED);
```

Comment:
This is just to fix some Oracle tests that sometimes fail due to the error:
```
ORA-00604: error occurred at recursive SQL level 1ORA-08177: can't serialize access for this transaction
``

Reply:
Well this method is setting the transaction level as serializable which is the highest isolation level. What's the reasoning to set the transaction level for one of the connections to READ COMMITED?

Reply:
But why this fixes it? Why oracle can't have the two connections serialized?

Reply:
To be honest, I'm not sure about this. Do you think we should revert this?

Reply:
Reverted.